src/datadog/span_data.cpp

```diff
@@ -51,9 +51,7 @@ void SpanData::apply_config(const SpanDefaults& defaults,
     tags.insert_or_assign(tags::version, version);
   }
   for (const auto& [key, value] : config.tags) {
-    if (!tags::is_internal(key)) {
-      tags.insert_or_assign(key, value);
-    }
```

Comment:
Do you want any user of the library to be able to set internal tags? Same question in `span.cpp`.

If Datadog has an internal client who wishes to have special access to the span representation, then consider adding an API specific to that use case, and document it as such.

Reply:
I'm not sure there is much value in forbidding internal tags, however I'm open to the idea of separating the `*_tag` and `*_metric` methods into `*_internal_tag` and `*internal_metric` if that makes it more palatable.